runtime/src/main/java/org/corfudb/runtime/exceptions/NetworkException.java

```diff
@@ -9,16 +9,28 @@
 public class NetworkException extends RuntimeException {
 
     @Getter
-    NodeLocator node;
+    private final NodeLocator node;
 
+    @Getter
+    private final Type exceptionType;
+
+    public static NetworkException buildDisconnected(NodeLocator node){
+        return new NetworkException(Type.DISCONNECTED, node);
+    }
 
-    public NetworkException(String message, NodeLocator node) {
-        super(message + " [endpoint=" + node.toString() + "]");
+    public NetworkException(Type exceptionType, NodeLocator node) {
+        super(exceptionType.name() + " [endpoint=" + node.toString() + "]");
+        this.exceptionType = exceptionType;
         this.node = node;
     }
 
-    public NetworkException(String message, NodeLocator node, Throwable cause) {
+    public NetworkException(Type exceptionType, String message, NodeLocator node, Throwable cause) {
         super(message + " [endpoint=" + node.toString() + "]", cause);
+        this.exceptionType = exceptionType;
         this.node = node;
     }
+
+    public enum Type {
+        DISCONNECTED
```

Comment:
What are the other types you think we can have?
Doesn't NetworkException imply disconnected?